src/effects/VST3/VST3EffectsModule.cpp

```diff
@@ -0,0 +1,296 @@
+/**********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  @file VST3EffectsModule.cpp
+
+  @author Vitaly Sverchinsky
+
+  @brief Part of Audacity VST3 module
+
+**********************************************************************/
+
+
+#include "VST3EffectsModule.h"
+
+#include <stdexcept>
+
+#include <wx/stdpaths.h>
+#include <wx/dir.h>
+#include <wx/log.h>
+#include <public.sdk/source/vst/hosting/module.h>
+
+#include "ModuleManager.h"
+
+#include "wxArrayStringEx.h"
+#include "PluginInterface.h"
+#include "ModuleInterface.h"
+#include "VST3Effect.h"
+#include "VST3Utils.h"
+#include "widgets/AButton.h"
+
+
+DECLARE_MODULE_ENTRY(AudacityModule)
+{
+   // Create our effects module and register
+   // Trust the module manager not to leak this
+   return safenew VST3EffectsModule();
+}
+
+DECLARE_BUILTIN_MODULE(VST3Builtin);
+
+//Helper class used to find vst3 effects during folder traversal
+class VST3PluginTraverser final : public wxDirTraverser
+{
+   std::function<void(const wxString&)> mOnPluginFound;
+public:
+   VST3PluginTraverser(std::function<void(const wxString&)> onPluginFound)
+      : mOnPluginFound(std::move(onPluginFound))
+   {
+   }
+
+   wxDirTraverseResult OnFile(const wxString& filename) override
+   {
+      if(filename.Matches("*.vst3"))
+         mOnPluginFound(filename);
+      return wxDIR_CONTINUE;
+   }
+   wxDirTraverseResult OnDir(const wxString& dirname) override
+   {
+      if(dirname.Matches("*.vst3"))
+      {
+         mOnPluginFound(dirname);
+         return wxDIR_IGNORE;//do not look inside...
+      }
+      return wxDIR_CONTINUE;
+   }
+};
+
+std::shared_ptr<VST3::Hosting::Module> VST3EffectsModule::GetModule(const wxString& path)
+{
+   const auto it = mModules.find(path);
+   if(it != mModules.end())
+   {
+      if(auto lock = it->second.lock())
+         return lock;
+   }
+
+   std::string moduleCreateError;
+   //VST sdk provides platform-specific module loading routines as well,
+   //implementation is condionally included (see CMakeLists.txt)
+   auto module = VST3::Hosting::Module::create(path.ToStdString(), moduleCreateError);
+   if(!module)
+      throw std::runtime_error(moduleCreateError.c_str());
+
+   mModules[path] = module;
+   return module;
+}
+
+PluginPath VST3EffectsModule::GetPath()
+{
+   return {};
+}
+
+ComponentInterfaceSymbol VST3EffectsModule::GetSymbol()
+{
+   return XO("VST3 Effects");
+}
+
+VendorSymbol VST3EffectsModule::GetVendor()
+{
+   return XO("The Audacity Team");
+}
+
+wxString VST3EffectsModule::GetVersion()
+{
+   return AUDACITY_VERSION_STRING;
+}
+
+TranslatableString VST3EffectsModule::GetDescription()
+{
+   return XO("Adds the ability to use VST3 effects in Audacity.");
+}
+
+bool VST3EffectsModule::Initialize()
+{
+   return true;
+}
+
+void VST3EffectsModule::Terminate()
+{
+}
+
+EffectFamilySymbol VST3EffectsModule::GetOptionalFamilySymbol()
+{
+   return XO("VST3");
```

Comment:
I see you broke the dependency cycle between VST3EffectsModule and VST3Effect.  The module depends on Effect, one-way.  You removed things from VST3EffectsModule.h and defined them in VST3Effect instead.

That is good, but now I see a duplication of this string constant.

Could you make a static function in VST3Effect.h and use it here?

.